### PR TITLE
chore: Add JSON formatter and format all JSON files in tests

### PR DESCRIPTION
**Pull request by aahung**
### Description of changes

Build our JSON formatter without prettier (node dependency)

```
...
reformatted tests/translator/output/aws-us-gov/version_deletion_policy_precedence.json
reformatted tests/translator/output/aws-us-gov/alexa_skill.json
reformatted tests/translator/output/aws-us-gov/connector_bucket_to_function.json
reformatted tests/translator/output/aws-us-gov/function_with_deployment_preference_multiple_combinations_conditions_with_passthrough.json
reformatted tests/feature_toggle/input/feature_toggle_config.json
1189 file(s) scanned.
1189 file(s) reformatted.
```

#### run it again:

```
./bin/json-format.py --check tests
1189 file(s) scanned.
0 file(s) need reformat.
```

### Checklist

- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by GavinZZ — APPROVED** [review]
Amazing!

File: bin/json-format.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                self.process_file(file_path)
+
+    def output_summary(self):
+        print(f"{self.scanned_file_found} file(s) scanned.")
+        if self.write:
+            print(f"{self.unformatted_file_count} file(s) reformatted.")
+        if self.check:
+            print(f"{self.unformatted_file_count} file(s) need reformat.")
+            if self.unformatted_file_count:
+                sys.exit(-1)
```
Comment (GavinZZ) [inline bin/json-format.py:62]: Why is this needed? The program will already exit after the print.

Comment (aahung) [reply bin/json-format.py:62]: -1 is an error code, so it will fail `make pr`

Comment (aahung) [reply bin/json-format.py:62]: https://stackoverflow.com/questions/44893807/i-want-to-know-what-exactly-sys-exit-1-returns-in-python

**Review by xazhao — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

**Review by hoffa — APPROVED** [review]
